From 44895b30b71be7b7e851ca04388d0f6d18f4aefd Mon Sep 17 00:00:00 2001
From: Michael Dickens <mlk@alum.mit.edu>
Date: Mon, 28 Oct 2013 20:58:46 -0400
Subject: [PATCH 4/5] utils: fix declaration of "env_path_sep" such that it is
 always initialized before it is used in the "get_env_paths" function, by
 moving it from the global scope to inside that function.  This change allows
 UHD_STATIC_BLOCK(load_modules) to work correctly.

---
 host/lib/utils/paths.cpp | 21 +++++++++++----------
 1 file changed, 11 insertions(+), 10 deletions(-)

diff --git a/host/lib/utils/paths.cpp b/host/lib/utils/paths.cpp
index 26fa6d1..6e0dd13 100644
--- a/host/lib/utils/paths.cpp
+++ b/host/lib/utils/paths.cpp
@@ -34,8 +34,19 @@
 namespace fs = boost::filesystem;
 
 /***********************************************************************
+ * Get a list of paths for an environment variable
+ **********************************************************************/
+static std::string get_env_var(const std::string &var_name, const std::string &def_val = ""){
+    const char *var_value_ptr = std::getenv(var_name.c_str());
+    return (var_value_ptr == NULL)? def_val : var_value_ptr;
+}
+
+static std::vector<fs::path> get_env_paths(const std::string &var_name){
+
+/***********************************************************************
  * Determine the paths separator
  **********************************************************************/
+
 #ifdef UHD_PLATFORM_WIN32
     static const std::string env_path_sep = ";";
 #else
@@ -46,16 +57,6 @@ namespace fs = boost::filesystem;
     boost::tokenizer<boost::char_separator<char> > \
     (inp, boost::char_separator<char>(env_path_sep.c_str()))
 
-/***********************************************************************
- * Get a list of paths for an environment variable
- **********************************************************************/
-static std::string get_env_var(const std::string &var_name, const std::string &def_val = ""){
-    const char *var_value_ptr = std::getenv(var_name.c_str());
-    return (var_value_ptr == NULL)? def_val : var_value_ptr;
-}
-
-static std::vector<fs::path> get_env_paths(const std::string &var_name){
-
     std::string var_value = get_env_var(var_name);
 
     //convert to filesystem path, filter blank paths
-- 
1.8.4.3

